### PR TITLE
(TWILL-69) Removed guava dependencies from twill-api and twill-common modules

### PR DESCRIPTION
**Pull request by chtyim**
Following changes are backward incompatible:
- TwillController no longer extends from Guava Service
  - No more start() method. stopAndWait() becomes terminate()
- TwillRunnerService no longer extends from Guava Service
  - Only has the blocking start() and stop() for start and stop.
- Classes moved from twill-common to twill-core, hence no long public
  - o.a.t.common.Services
  - o.a.t.common.ServiceListenerAdapter
  - o.a.t.common.CompositeService
  - o.a.t.common.DefaultResourceReport

Modified all unit-tests and examples to use the new API

### CODE REVIEW

File: twill-core/src/test/java/org/apache/twill/internal/ControllerTest.java [hunk 1]
```diff
@@ -191,7 +186,7 @@ protected void triggerShutdown() {
   }
 
   private TwillController getController(ZKClient zkClient, RunId runId) {
-    TwillController controller = new AbstractTwillController(runId, zkClient, ImmutableList.<LogHandler>of()) {
+    AbstractTwillController controller = new AbstractTwillController(runId, zkClient, ImmutableList.<LogHandler>of()) {
```
Comment (henry-cask) [inline twill-core/src/test/java/org/apache/twill/internal/ControllerTest.java]: Any reason why this change to AbstractTwillController?

File: twill-yarn/src/main/java/org/apache/twill/yarn/YarnTwillRunnerService.java [hunk 2]
```diff
@@ -408,26 +429,17 @@ public void cancel() {
...
+      public void run() {
         synchronized (YarnTwillRunnerService.this) {
           Iterables.removeIf(controllers.values(),
                              new Predicate<TwillController>() {
-             @Override
-             public boolean apply(TwillController input) {
-               return input == controller;
-             }
-           });
+                               @Override
```
Comment (henry-cask) [inline twill-yarn/src/main/java/org/apache/twill/yarn/YarnTwillRunnerService.java]: Accidental indents edits?

**Comment by henry-cask** [comment]
Just did a pass and post some questions.

File: twill-core/src/test/java/org/apache/twill/internal/ControllerTest.java (hunk 1 above)
Comment (chtyim) [inline twill-core/src/test/java/org/apache/twill/internal/ControllerTest.java]: The actual instance is an instance of `AbstractTwillController`. For the unit-test, we need to start the controller, which the start method only available in the `AbstractTwillController`.

File: twill-yarn/src/main/java/org/apache/twill/yarn/YarnTwillRunnerService.java (hunk 2 above)
Comment (chtyim) [inline twill-yarn/src/main/java/org/apache/twill/yarn/YarnTwillRunnerService.java]: oops, probably done by the IDE automatically. Will fix it.

**Comment by henry-cask** [comment]
+1 LGTM

**Comment by chtyim** [comment]
Merged.
